[ML] Fix UI inconsistencies in APM Failed transaction correlations

qn895 opened the pull request:
## Summary

This PR addresses the issues from #109061 / #109093 / #106381

<img width="1028" alt="Screen Shot 2021-08-18 at 18 39 47" src="https://user-images.githubusercontent.com/43350163/129985766-373279ff-79dc-4001-8d06-3efe9a1b2421.png">
<img width="1028" alt="Screen Shot 2021-08-18 at 18 39 50" src="https://user-images.githubusercontent.com/43350163/129985769-636fc054-a7f3-4332-93ca-949444e45aa2.png">

- [x] Show the same empty state in the correlations table
<img width="1028" alt="Screen Shot 2021-08-18 at 20 57 44" src="https://user-images.githubusercontent.com/43350163/129995613-a86c3af7-3440-4c6b-b3a4-e8ee86ebac53.png">

- [x] Add "Correlations" title above the table
- [x] Add `EuiSpacer` between the sections before and after progress
- [x] Update the copy within the beta badge
`title=Failed transaction correlations` `description=Failed transaction correlations is not GA...`
- [x] Remove `s` size from the beta badge
- [x] Move the help popover to the top of the panel (similar to the Latency correlations tab)
- [x] Move the Cancel/Refresh option to the right of the progress bar (similar to the Latency correlations tab)
- [x] When the correlation job is running the correlations tab should show a loading state similar to the latency correlations table
- [x] Indicate in the table headers Score is sorted by default
- [x] Add sortability to both Latency and failed transactions correlations table 
- [x] Refactor to prevent duplicate code/components like Log, progress bar
- [x] Fix alignments of the tab content header (previously navigating from one Trace samples tab to Latency correlation tabs will cause a minor jump in the header, or the titles within the same tab were not the same size ) 
- [x] Remove the `event.outcome` as a field candidate (because `event.outcome: failure` would always be significant in this case)
- [x] Shrink the column width for impact (previously it was at 116px)
- [x] Added badge for High, medium, low #109093
<img width="1468" alt="Screen Shot 2021-08-19 at 11 53 11" src="https://user-images.githubusercontent.com/43350163/130111156-403346d3-5136-41ed-90dc-28dcbc203271.png">

- [x] Fix license prompt text
<img width="1000" alt="Screen Shot 2021-08-19 at 16 16 06" src="https://user-images.githubusercontent.com/43350163/130145316-afbd5031-e508-475c-90d1-7433c6ada6a8.png">
- [x] Functional tests for the new tab 
- [x] Make the p value & error rate columns visible only when esInspect mode is enabled

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)

qn895 committed:
[ML] Fix UI inconsistencies, add sorting

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/)
* Commit: 8dade28f1bee474b913562a74a593e0b7d7ec50d
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/8dade28f1bee474b913562a74a593e0b7d7ec50d)
* [Documentation Changes](https://kibana_109187.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f719aa40319d316616069ba67c4ad921eeccc531) | [after](https://github.com/elastic/kibana/commit/8dade28f1bee474b913562a74a593e0b7d7ec50d) | diff |
| --- | --- | --- | --- |
| `apm` | 1611 | 1615 | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f719aa40319d316616069ba67c4ad921eeccc531) | [after](https://github.com/elastic/kibana/commit/8dade28f1bee474b913562a74a593e0b7d7ec50d) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | -305.0B |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/","number":"146803","commit":"8dade28f1bee474b913562a74a593e0b7d7ec50d"}]}
PIPELINE-->

qn895 committed:
[ML] Fix columns for latency sortable

qn895 committed:
[ML] Fix columns for latency sortable

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/)
* Commit: a145372bfe3139f2b3efaf40343415aa8393e68a
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/execution/node/416/log)

### History
* :green_heart: [Build #146803](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/) succeeded 8dade28f1bee474b913562a74a593e0b7d7ec50d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/","number":"146803","commit":"8dade28f1bee474b913562a74a593e0b7d7ec50d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/","number":"146817","commit":"a145372bfe3139f2b3efaf40343415aa8393e68a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/)
* Commit: 420e1831a52ea4748d2fc28efb34c59877306a80
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/420e1831a52ea4748d2fc28efb34c59877306a80)
* [Documentation Changes](https://kibana_109187.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f719aa40319d316616069ba67c4ad921eeccc531) | [after](https://github.com/elastic/kibana/commit/420e1831a52ea4748d2fc28efb34c59877306a80) | diff |
| --- | --- | --- | --- |
| `apm` | 1611 | 1615 | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f719aa40319d316616069ba67c4ad921eeccc531) | [after](https://github.com/elastic/kibana/commit/420e1831a52ea4748d2fc28efb34c59877306a80) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | -37.0B |

### History
* :broken_heart: [Build #146817](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/) failed a145372bfe3139f2b3efaf40343415aa8393e68a
* :green_heart: [Build #146803](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/) succeeded 8dade28f1bee474b913562a74a593e0b7d7ec50d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/","number":"146803","commit":"8dade28f1bee474b913562a74a593e0b7d7ec50d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/","number":"146817","commit":"a145372bfe3139f2b3efaf40343415aa8393e68a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/","number":"146819","commit":"420e1831a52ea4748d2fc28efb34c59877306a80"}]}
PIPELINE-->

formgeist reviewed (approved):
Overall everything looks great! 👏 Just a few copy additions/changes suggested.

walterra commented:
I noticed when trying to sort the Impact-column low/medium/high gets sorted alphabetically. Is there a way we can sort it by the underlying impact threshold?

For example, here I'd expect it to sort by descending impact (high/medium/low) instead of descending alphabetically:

![image](https://user-images.githubusercontent.com/230104/130060096-4c8a26f4-daca-41e1-ac31-9ff8a2d7e0a6.png)

walterra commented on x-pack/plugins/apm/public/components/app/transaction_details/distribution/index.tsx:
```diff
@@ -132,7 +132,7 @@ export function TransactionDistribution({
   return (
     <div data-test-subj="apmTransactionDistributionTabContent">
-      <EuiFlexGroup>
+      <EuiFlexGroup style={{ minHeight: 56 }}>
```
Can you explain why this is necessary? Could we use an EUI px var instead?

qn895 replied:
Added a comment for it [here](https://github.com/elastic/kibana/pull/109187/commits/be2657501e18cd41c101492c3af014f0c56c4ee2). The small size selection text is causing the width of that title box to be smaller than the other two (which have the help button), so when you navigate between the tabs there's a bit of a "flickering" or shifting of the title/header spacer.

walterra commented on x-pack/plugins/apm/public/components/app/correlations/log.tsx:
```diff
@@ -0,0 +1,38 @@
+/*
```
Nit: Suggest to name this file `correlations_log.tsx` for consistency.

qn895 replied:
Renamed here https://github.com/elastic/kibana/pull/109187/commits/6f9ffa172004850bd29b2eae35d4740b74aa040c

qn895 committed:
[ML] Rename log -> correlations_log.tsx, update help texts

qn895 committed:
[ML] Remove event.outcome as candidate for failed transaction correlation

qn895 committed:
[ML] Shrink the score column

qn895 committed:
[ML] Add colored badge for high, medium, low thresholds

qn895 committed:
[ML] Comment on hard coded height

qn895 committed:
[ML] Sort by normalized

qn895 commented on x-pack/plugins/apm/public/components/app/correlations/failed_transactions_correlations_help_popover.tsx:
```diff
@@ -18,32 +18,36 @@ export function FailedTransactionsCorrelationsHelpPopover() {
+      title={i18n.translate(
+        'xpack.apm.correlations.failedTransactions.helpPopover.title',
+        {
+          defaultMessage: 'Failed transactions correlations',
```
Updated here https://github.com/elastic/kibana/pull/109187/commits/6f9ffa172004850bd29b2eae35d4740b74aa040c

qn895 commented on x-pack/plugins/apm/public/components/app/correlations/failed_transactions_correlations_help_popover.tsx:
```diff
@@ -18,32 +18,36 @@ export function FailedTransactionsCorrelationsHelpPopover() {
         <FormattedMessage
-          id="xpack.apm.correlations.failurePopoverBasicExplanation"
+          id="xpack.apm.correlations.failedTransactions.helpPopover.basicExplanation"
           defaultMessage="Correlations help you discover which attributes are contributing to failed transactions."
```
Updated here https://github.com/elastic/kibana/pull/109187/commits/6f9ffa172004850bd29b2eae35d4740b74aa040c

qn895 commented:
> I noticed when trying to sort the Impact-column low/medium/high gets sorted alphabetically. Is there a way we can sort it by the underlying impact threshold?

Thanks for caching that. Updated here https://github.com/elastic/kibana/pull/109187/commits/84a0514986f39d349c6e35fbf2cfe9a68588a573

Latest changes also include:
- Remove the `event.outcome` as a field candidate (because `event.outcome: failure` would always be significant in this case)
- Shrink the column width for impact (previously it was at 116px)
- Added badge for High, medium, low https://github.com/elastic/kibana/issues/109093

qn895 committed:
[ML] Fix latency encoding

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/)
* Commit: 84a0514986f39d349c6e35fbf2cfe9a68588a573
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/84a0514986f39d349c6e35fbf2cfe9a68588a573)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/execution/node/172/log)

### History
* :green_heart: [Build #146819](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/) succeeded 420e1831a52ea4748d2fc28efb34c59877306a80
* :broken_heart: [Build #146817](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/) failed a145372bfe3139f2b3efaf40343415aa8393e68a
* :green_heart: [Build #146803](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/) succeeded 8dade28f1bee474b913562a74a593e0b7d7ec50d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/","number":"146803","commit":"8dade28f1bee474b913562a74a593e0b7d7ec50d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/","number":"146817","commit":"a145372bfe3139f2b3efaf40343415aa8393e68a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/","number":"146819","commit":"420e1831a52ea4748d2fc28efb34c59877306a80"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/","number":"146998","commit":"84a0514986f39d349c6e35fbf2cfe9a68588a573"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/)
* Commit: 3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/execution/node/187/log)

### History
* :broken_heart: [Build #146998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/) failed 84a0514986f39d349c6e35fbf2cfe9a68588a573
* :green_heart: [Build #146819](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/) succeeded 420e1831a52ea4748d2fc28efb34c59877306a80
* :broken_heart: [Build #146817](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/) failed a145372bfe3139f2b3efaf40343415aa8393e68a
* :green_heart: [Build #146803](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/) succeeded 8dade28f1bee474b913562a74a593e0b7d7ec50d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/","number":"146803","commit":"8dade28f1bee474b913562a74a593e0b7d7ec50d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/","number":"146817","commit":"a145372bfe3139f2b3efaf40343415aa8393e68a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/","number":"146819","commit":"420e1831a52ea4748d2fc28efb34c59877306a80"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/","number":"146998","commit":"84a0514986f39d349c6e35fbf2cfe9a68588a573"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/","number":"147023","commit":"3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd"}]}
PIPELINE-->

qn895 committed:
[ML] Add functional tests

qn895 committed:
[ML] Update jest tests

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/)
* Commit: 8300462f4a8bc1ad83dc40138a529b33b2a0d624
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/8300462f4a8bc1ad83dc40138a529b33b2a0d624)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/execution/node/675/log)
* [Execute xpack-kibana-ciGroup6](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/execution/node/839/log)

### History
* :broken_heart: [Build #146998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/) failed 84a0514986f39d349c6e35fbf2cfe9a68588a573
* :broken_heart: [Build #147023](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/) failed 3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd
* :green_heart: [Build #146819](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/) succeeded 420e1831a52ea4748d2fc28efb34c59877306a80
* :broken_heart: [Build #146817](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/) failed a145372bfe3139f2b3efaf40343415aa8393e68a
* :green_heart: [Build #146803](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/) succeeded 8dade28f1bee474b913562a74a593e0b7d7ec50d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146803/","number":"146803","commit":"8dade28f1bee474b913562a74a593e0b7d7ec50d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/","number":"146817","commit":"a145372bfe3139f2b3efaf40343415aa8393e68a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/","number":"146819","commit":"420e1831a52ea4748d2fc28efb34c59877306a80"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/","number":"147023","commit":"3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/","number":"146998","commit":"84a0514986f39d349c6e35fbf2cfe9a68588a573"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/","number":"147075","commit":"8300462f4a8bc1ad83dc40138a529b33b2a0d624"}]}
PIPELINE-->

qn895 committed:
[ML] Fix license text, test subj

sqren commented on x-pack/plugins/apm/common/search_strategies/failure_correlations/constants.ts:
```diff
@@ -30,3 +30,9 @@ export const FAILED_TRANSACTIONS_IMPACT_THRESHOLD = {
 } as const;
+
+export const FAILED_TRANSACTIONS_IMPACT_THRESHOLD_COLOR = {
+  HIGH: '#f38967',
+  MEDIUM: '#f4d872',
+  LOW: '#dee5ec',
```
You should get these from Eui

qn895 replied:
Thanks for catching that. This was not used and should have been removed. Removed here https://github.com/elastic/kibana/pull/109187/commits/85caf980c6e6e4d8179772974ac8723fbff65171

qn895 committed:
[ML] Remove unused color constant

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/)
* Commit: 1497d3d7fd09756253e6bee89d0aa9a954d300f9
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/1497d3d7fd09756253e6bee89d0aa9a954d300f9)
* [Documentation Changes](https://kibana_109187.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f719aa40319d316616069ba67c4ad921eeccc531) | [after](https://github.com/elastic/kibana/commit/1497d3d7fd09756253e6bee89d0aa9a954d300f9) | diff |
| --- | --- | --- | --- |
| `apm` | 1611 | 1615 | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f719aa40319d316616069ba67c4ad921eeccc531) | [after](https://github.com/elastic/kibana/commit/1497d3d7fd09756253e6bee89d0aa9a954d300f9) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +144.0B |

### History
* :broken_heart: [Build #147075](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/) failed 8300462f4a8bc1ad83dc40138a529b33b2a0d624
* :broken_heart: [Build #146998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/) failed 84a0514986f39d349c6e35fbf2cfe9a68588a573
* :broken_heart: [Build #147023](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/) failed 3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd
* :green_heart: [Build #146819](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/) succeeded 420e1831a52ea4748d2fc28efb34c59877306a80
* :broken_heart: [Build #146817](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/) failed a145372bfe3139f2b3efaf40343415aa8393e68a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146817/","number":"146817","commit":"a145372bfe3139f2b3efaf40343415aa8393e68a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/","number":"146819","commit":"420e1831a52ea4748d2fc28efb34c59877306a80"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/","number":"147023","commit":"3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/","number":"146998","commit":"84a0514986f39d349c6e35fbf2cfe9a68588a573"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/","number":"147075","commit":"8300462f4a8bc1ad83dc40138a529b33b2a0d624"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/","number":"147133","commit":"1497d3d7fd09756253e6bee89d0aa9a954d300f9"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/)
* Commit: 85caf980c6e6e4d8179772974ac8723fbff65171
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/85caf980c6e6e4d8179772974ac8723fbff65171)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/execution/node/675/log)
* [Execute xpack-kibana-ciGroup4](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/execution/node/782/log)

### History
* :green_heart: [Build #147133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/) succeeded 1497d3d7fd09756253e6bee89d0aa9a954d300f9
* :broken_heart: [Build #147075](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/) failed 8300462f4a8bc1ad83dc40138a529b33b2a0d624
* :broken_heart: [Build #146998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/) failed 84a0514986f39d349c6e35fbf2cfe9a68588a573
* :broken_heart: [Build #147023](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/) failed 3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd
* :green_heart: [Build #146819](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/) succeeded 420e1831a52ea4748d2fc28efb34c59877306a80

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146819/","number":"146819","commit":"420e1831a52ea4748d2fc28efb34c59877306a80"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/","number":"147023","commit":"3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/","number":"146998","commit":"84a0514986f39d349c6e35fbf2cfe9a68588a573"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/","number":"147075","commit":"8300462f4a8bc1ad83dc40138a529b33b2a0d624"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/","number":"147133","commit":"1497d3d7fd09756253e6bee89d0aa9a954d300f9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/","number":"147181","commit":"85caf980c6e6e4d8179772974ac8723fbff65171"}]}
PIPELINE-->

formgeist commented:
> * Added badge for High, medium, low [[APM] Correlations: Show impact levels (high, medium, low) as colored badges indicating their severity #109093](https://github.com/elastic/kibana/issues/109093)
> 
> <img alt="Screen Shot 2021-08-19 at 11 53 11" width="1468" src="https://user-images.githubusercontent.com/43350163/130111156-403346d3-5136-41ed-90dc-28dcbc203271.png">

@qn895 Thanks for adding those as well - much nicer visual reference now for the user 👍

sqren commented on x-pack/plugins/apm/public/components/app/correlations/correlations_log.tsx:
```diff
@@ -0,0 +1,38 @@
+    >
+      <EuiPanel color="subdued">
+        {log.map((d, i) => {
+          const splitItem = d.split(': ');
```
This looks odd. Instead of splitting, can we return logs in a structured format? It looks like we concatenate them here:

https://github.com/elastic/kibana/blob/d155a94fcec95cb6f99e879b099652623fc80451/x-pack/plugins/apm/server/lib/search_strategies/correlations/async_search_service_log.ts#L28

I have some other minor suggestions for that file:
```ts
import { currentTimeAsString } from './utils';

interface LogMessage {
  timestamp: string;
  message: string;
  error?: string;
}

export const asyncSearchServiceLogProvider = () => {
  const logMessages: LogMessage[] = [];

  function addLogMessage(message: string, error?: string) {
    logMessages.push({ timestamp: currentTimeAsString(), message, error });
  }

  function getLogMessages() {
    return logMessages;
  }

  return { addLogMessage, getLogMessages };
};

export type AsyncSearchServiceLog = ReturnType<
  typeof asyncSearchServiceLogProvider
>;
```

sqren replied:
While diving into the above log stuff I noticed this section:

https://github.com/elastic/kibana/blob/d155a94fcec95cb6f99e879b099652623fc80451/x-pack/plugins/apm/server/lib/search_strategies/correlations/search_strategy.ts#L55-L64

That can be simplified to:
```ts
      if (request.id) {
        const existingGetAsyncSearchServiceState = asyncSearchServiceMap.get(
          request.id
        );

        if (existingGetAsyncSearchServiceState === undefined) {
          throw new Error(
            `AsyncSearchService with ID '${request.id}' does not exist.`
          );
        }
```

walterra replied:
This code/logic didn't change since 7.14, it was just moved to a separate component here to consolidate duplicate code. Moving to a structured log would require touching both server+client code again. Since we're past FF I'd like to suggest to leave it as is for 7.15. We already have an item for 7.16 to revisit this (`Log Error handling ...` item) here #109220

sqren replied:
Doing this for 7.16 is fine. Do you want to track this by referencing this comment? 

I also recall adding a comment about replacing the generator function with promises (preferably) or observables since that's what we already use and we should keep the number of async concepts down when possible. Can we add that to the list as well?

walterra replied:
👍  I linked back to this conversation from the meta issue's item. In 7.15 we moved the generator function to it's own file and added tests so it should be more safe to refactor from now on. I'll add an item about it too.

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

sqren commented on x-pack/plugins/apm/public/components/app/correlations/correlations_log.tsx:
```diff
@@ -0,0 +1,38 @@
+      })}
+    >
+      <EuiPanel color="subdued">
+        {log.map((d, i) => {
```
```suggestion
        {logMessages.map((logMessage, i) => {
```

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/109187/commits/82d5457f4696584969059f9ad78a897722507d2c

sqren commented on x-pack/plugins/apm/public/components/app/correlations/correlations_log.tsx:
```diff
@@ -0,0 +1,38 @@
+          return (
+            <p key={i}>
+              <small>
+                <EuiCode>{splitItem[0]}</EuiCode> {splitItem[1]}
```
```suggestion
                <EuiCode>{asAbsoluteDateTime(logMessage.timestamp)}</EuiCode> {logMessage.message}
```

sqren replied:
Btw I just noticed that we need a few tests for `asAbsoluteDateTime`. Can you added the following to `x-pack/plugins/apm/common/utils/formatters/datetime.test.ts`

```tsx
    it('milliseconds', () => {
      moment.tz.setDefault('Europe/Copenhagen');
      expect(asAbsoluteDateTime(1559390400000, 'milliseconds')).toBe(
        'Jun 1, 2019, 14:00:00.000 (UTC+2)'
      );
    });

    it('seconds', () => {
      moment.tz.setDefault('Europe/Copenhagen');
      expect(asAbsoluteDateTime(1559390400000, 'seconds')).toBe(
        'Jun 1, 2019, 14:00:00 (UTC+2)'
      );
    });

    it('minutes', () => {
      moment.tz.setDefault('Europe/Copenhagen');
      expect(asAbsoluteDateTime(1559390400000, 'minutes')).toBe(
        'Jun 1, 2019, 14:00 (UTC+2)'
      );
    });

    it('hours', () => {
      moment.tz.setDefault('Europe/Copenhagen');
      expect(asAbsoluteDateTime(1559390400000, 'hours')).toBe(
        'Jun 1, 2019, 14 (UTC+2)'
      );
    });
```

Thanks!

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/109187/commits/82d5457f4696584969059f9ad78a897722507d2c

sqren commented on x-pack/plugins/apm/public/components/app/correlations/empty_state_prompt.tsx:
```diff
@@ -0,0 +1,48 @@
+          </EuiText>
+        }
+        body={
+          <>
+            <EuiText size="s">
+              <FormattedMessage
+                id="xpack.apm.correlations.noCorrelationsTextLine1"
+                defaultMessage="Correlations will only be identified if they have significant impact."
+              />
+            </EuiText>
+            {/* Another EuiText element to enforce a line break */}
+            <EuiText size="s">
+              <FormattedMessage
+                id="xpack.apm.correlations.noCorrelationsTextLine2"
+                defaultMessage="Try selecting another time range or remove any added filter."
+              />
+            </EuiText>
+          </>
```
It sounds like you want two paragraphs:

```suggestion
          <EuiText size="s">
            <p>
              {i18n.translate("xpack.apm.correlations.noCorrelationsTextLine1", {
                defaultMessage:
                  "Correlations will only be identified if they have significant impact.",
              })}
            </p>

            <p>
              {i18n.translate("xpack.apm.correlations.noCorrelationsTextLine2", {
                defaultMessage:
                  "Try selecting another time range or remove any added filter.",
              })}
            </p>
          </EuiText>
```

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/109187/commits/82d5457f4696584969059f9ad78a897722507d2c

sqren commented on x-pack/plugins/apm/public/components/app/correlations/latency_correlations.tsx:
```diff
@@ -256,17 +253,48 @@ export function LatencyCorrelations({ onFilter }: { onFilter: () => void }) {
+    if (!Array.isArray(histograms)) {
+      return { histogramTerms: [], sorting: undefined };
+    }
+    const sortedTerms = sortBy(
+      histograms.map((d) => {
+        return {
+          fieldName: d.field,
+          fieldValue: d.value,
+          ksTest: d.ksTest,
+          correlation: d.correlation,
+          duplicatedFields: d.duplicatedFields,
+        };
+      }),
+      sortField
+    );
+
+    return {
+      histogramTerms:
+        sortDirection === 'asc'
+          ? sortedTerms
+          : (sortedTerms.reverse() as MlCorrelationsTerms[]),
```
You can use `orderBy` instead:

```suggestion
    const sortedTerms = orderBy(
      histograms.map((d) => {
        return {
          fieldName: d.field,
          fieldValue: d.value,
          ksTest: d.ksTest,
          correlation: d.correlation,
          duplicatedFields: d.duplicatedFields,
        };
      }),
      sortField,
      sortDirection
    );

    return {
      histogramTerms: sortedTerms,
```

qn895 replied:
Updated here https://github.com/elastic/kibana/pull/109187/commits/82d5457f4696584969059f9ad78a897722507d2c

sqren commented on x-pack/plugins/apm/public/components/app/transaction_details/distribution/index.tsx:
```diff
@@ -27,6 +27,9 @@ import { useApmParams } from '../../../../hooks/use_apm_params';
 const DEFAULT_PERCENTILE_THRESHOLD = 95;
+// Enforce min height so it's consistent across all tabs on the same level
+// to prevent "flickering" behavior
+const MIN_TAB_TITLE_HEIGHT = 56;
```
Why do the tabs change height in the first place?

qn895 replied:
The small size `selection text` prompt is causing the width of that title box to be smaller than the other two (which have the standard size help button), so when you navigate between the tabs there's a bit of a "flickering" or shifting of the title/header spacer.

sqren commented on x-pack/plugins/apm/server/lib/search_strategies/failed_transactions_correlations/async_search_service.ts:
```diff
@@ -35,10 +36,11 @@ export const asyncErrorCorrelationSearchServiceProvider = (
+        fieldCandidates: candidates,
+      } = await fetchTransactionDurationFieldCandidates(esClient, params);
+
+      const fieldCandidates = candidates.filter((t) => !(t === EVENT_OUTCOME));
```
Can you move this transformation to `fetchTransactionDurationFieldCandidates`?

```suggestion

```

Additionally, since we eventually want this to be an array, can we do away with the `Set` and just use make it an array from the beginning. Currently we use `forEach` and mutate two separate data structures  (`acceptableFields` and `finalFieldCandidates`). I think we can simplify this by using immutable operation (`filter` and `map`) and perhaps only a single data structure.

sqren commented on x-pack/test/functional/apps/apm/correlations/failed_transaction_correlations.ts:
```diff
@@ -0,0 +1,155 @@
+      });
+    });
+  });
+}
```
Thanks for adding this test! Do we already have API tests for the correlations endpoint? If not, can you please add one? Thanks!

qn895 replied:
Added the test for the failed transaction correlations search strategy here https://github.com/elastic/kibana/pull/109187/commits/e683e57ee486a81512223cff873b35730b0346a3

qn895 committed:
Merge remote-tracking branch 'upstream/master' into ml-fix-apm-transactions-ux

qn895 committed:
[ML] Update correlations_log, add more tests for asAbsoluteDateTime, use orderBy

qn895 committed:
[ML] Only show the p-value hover on inspect mode

qn895 committed:
[ML] Add new error rate columns

qn895 committed:
Merge remote-tracking branch 'upstream/master' into ml-fix-apm-transactions-ux

qn895 committed:
[ML] Add new error rate columns

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/)
* Commit: 64036160b6b4ce18eaa54cea755c100e94940651
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/64036160b6b4ce18eaa54cea755c100e94940651)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/execution/node/531/log)
* [Execute xpack-APMCypress](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/execution/node/740/log)

### History
* :broken_heart: [Build #147181](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/) failed 85caf980c6e6e4d8179772974ac8723fbff65171
* :green_heart: [Build #147133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/) succeeded 1497d3d7fd09756253e6bee89d0aa9a954d300f9
* :broken_heart: [Build #147075](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/) failed 8300462f4a8bc1ad83dc40138a529b33b2a0d624
* :broken_heart: [Build #146998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/) failed 84a0514986f39d349c6e35fbf2cfe9a68588a573
* :broken_heart: [Build #147023](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/) failed 3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147023/","number":"147023","commit":"3e34c5a77d2fb7c7ecadbc1f1f8db3e4ec08e2cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/","number":"146998","commit":"84a0514986f39d349c6e35fbf2cfe9a68588a573"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/","number":"147075","commit":"8300462f4a8bc1ad83dc40138a529b33b2a0d624"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/","number":"147133","commit":"1497d3d7fd09756253e6bee89d0aa9a954d300f9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/","number":"147181","commit":"85caf980c6e6e4d8179772974ac8723fbff65171"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/","number":"147520","commit":"64036160b6b4ce18eaa54cea755c100e94940651"}]}
PIPELINE-->

qn895 committed:
[ML] Add api integration test for failed transaction ss

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147535/)
* Commit: e683e57ee486a81512223cff873b35730b0346a3
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/e683e57ee486a81512223cff873b35730b0346a3)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147535/execution/node/550/log)
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147535/execution/node/951/log)

### History
* :broken_heart: [Build #147520](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/) failed 64036160b6b4ce18eaa54cea755c100e94940651
* :broken_heart: [Build #147181](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/) failed 85caf980c6e6e4d8179772974ac8723fbff65171
* :green_heart: [Build #147133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/) succeeded 1497d3d7fd09756253e6bee89d0aa9a954d300f9
* :broken_heart: [Build #147075](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/) failed 8300462f4a8bc1ad83dc40138a529b33b2a0d624
* :broken_heart: [Build #146998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/) failed 84a0514986f39d349c6e35fbf2cfe9a68588a573

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/146998/","number":"146998","commit":"84a0514986f39d349c6e35fbf2cfe9a68588a573"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/","number":"147075","commit":"8300462f4a8bc1ad83dc40138a529b33b2a0d624"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/","number":"147133","commit":"1497d3d7fd09756253e6bee89d0aa9a954d300f9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/","number":"147181","commit":"85caf980c6e6e4d8179772974ac8723fbff65171"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/","number":"147520","commit":"64036160b6b4ce18eaa54cea755c100e94940651"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147535/","number":"147535","commit":"e683e57ee486a81512223cff873b35730b0346a3"}]}
PIPELINE-->

qn895 committed:
[ML] Fix api tests

qn895 committed:
Merge remote-tracking branch 'upstream/master' into ml-fix-apm-transactions-ux

sqren commented on x-pack/test/apm_api_integration/tests/correlations/failed_transactions.ts:
```diff
@@ -0,0 +1,242 @@
+    };
+  };
+
+  registry.when('quynh on trial license without data', { config: 'trial', archives: [] }, () => {
```
Just testing stuff out with your own name I assume :D 

```suggestion
  registry.when('on trial license without data', { config: 'trial', archives: [] }, () => {
```

walterra replied:
Fixed in 909e203 😅

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147587/)
* Commit: e4c0d3ad1a325a4ac6f0864d5637b9c03c215dc5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/e4c0d3ad1a325a4ac6f0864d5637b9c03c215dc5)
* [Documentation Changes](https://kibana_109187.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/08e47c8dd388f4c7f5f10ce9fedb7f6994cd6d29) | [after](https://github.com/elastic/kibana/commit/e4c0d3ad1a325a4ac6f0864d5637b9c03c215dc5) | diff |
| --- | --- | --- | --- |
| `apm` | 1611 | 1615 | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/08e47c8dd388f4c7f5f10ce9fedb7f6994cd6d29) | [after](https://github.com/elastic/kibana/commit/e4c0d3ad1a325a4ac6f0864d5637b9c03c215dc5) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +2.7KB |

### History
* :broken_heart: [Build #147535](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147535/) failed e683e57ee486a81512223cff873b35730b0346a3
* :broken_heart: [Build #147520](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/) failed 64036160b6b4ce18eaa54cea755c100e94940651
* :broken_heart: [Build #147181](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/) failed 85caf980c6e6e4d8179772974ac8723fbff65171
* :green_heart: [Build #147133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/) succeeded 1497d3d7fd09756253e6bee89d0aa9a954d300f9
* :broken_heart: [Build #147075](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/) failed 8300462f4a8bc1ad83dc40138a529b33b2a0d624

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147075/","number":"147075","commit":"8300462f4a8bc1ad83dc40138a529b33b2a0d624"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/","number":"147133","commit":"1497d3d7fd09756253e6bee89d0aa9a954d300f9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/","number":"147181","commit":"85caf980c6e6e4d8179772974ac8723fbff65171"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/","number":"147520","commit":"64036160b6b4ce18eaa54cea755c100e94940651"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147535/","number":"147535","commit":"e683e57ee486a81512223cff873b35730b0346a3"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147587/","number":"147587","commit":"e4c0d3ad1a325a4ac6f0864d5637b9c03c215dc5"}]}
PIPELINE-->

walterra committed:
fix api test names

walterra reviewed (approved):
Latest changes LGTM

sqren commented on x-pack/test/apm_api_integration/tests/correlations/failed_transactions.ts:
```diff
@@ -0,0 +1,238 @@
+        'search strategy eventually completes and returns full results',
+        5000,
+        async () => {
+          const response = await supertest
+            .post(`/internal/bsearch`)
+            .set('kbn-xsrf', 'foo')
+            .send(reqBody);
+
+          followUpResponse = parseBfetchResponse(response)[0];
```
It looks like you always want to parse the responses made to the `/bsearch` endpoint. In that case, what about creating a bfetchClient which takes the request body and returns the (parsed) response?

```suggestion
          followUpResponse = await bfetchClient(supertest, reqBody);
```

You could also make bFetchClient a service similar to `supertestAsApmWriteUser` and then you can call it without specifying `supertest`:

```suggestion
		  const supertest = context.getService('bfetchClient');
          followUpResponse = await bfetchClient(reqBody);
```

walterra replied:
Moved this to an item on the 7.16 meta issue: #109220

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147646/)
* Commit: 909e20314ed9d21ac4966bc445d3f46f87cbb25e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-109187/909e20314ed9d21ac4966bc445d3f46f87cbb25e)
* [Documentation Changes](https://kibana_109187.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/08e47c8dd388f4c7f5f10ce9fedb7f6994cd6d29) | [after](https://github.com/elastic/kibana/commit/909e20314ed9d21ac4966bc445d3f46f87cbb25e) | diff |
| --- | --- | --- | --- |
| `apm` | 1611 | 1615 | +4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/08e47c8dd388f4c7f5f10ce9fedb7f6994cd6d29) | [after](https://github.com/elastic/kibana/commit/909e20314ed9d21ac4966bc445d3f46f87cbb25e) | diff |
| --- | --- | --- | --- |
| `apm` | 4.4MB | 4.4MB | +2.7KB |

### History
* :green_heart: [Build #147587](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147587/) succeeded e4c0d3ad1a325a4ac6f0864d5637b9c03c215dc5
* :broken_heart: [Build #147535](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147535/) failed e683e57ee486a81512223cff873b35730b0346a3
* :broken_heart: [Build #147520](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/) failed 64036160b6b4ce18eaa54cea755c100e94940651
* :broken_heart: [Build #147181](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/) failed 85caf980c6e6e4d8179772974ac8723fbff65171
* :green_heart: [Build #147133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/) succeeded 1497d3d7fd09756253e6bee89d0aa9a954d300f9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @qn895

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147133/","number":"147133","commit":"1497d3d7fd09756253e6bee89d0aa9a954d300f9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147181/","number":"147181","commit":"85caf980c6e6e4d8179772974ac8723fbff65171"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147520/","number":"147520","commit":"64036160b6b4ce18eaa54cea755c100e94940651"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147535/","number":"147535","commit":"e683e57ee486a81512223cff873b35730b0346a3"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147587/","number":"147587","commit":"e4c0d3ad1a325a4ac6f0864d5637b9c03c215dc5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/147646/","number":"147646","commit":"909e20314ed9d21ac4966bc445d3f46f87cbb25e"}]}
PIPELINE-->

kibanamachine commented:
## 💚 Backport successful

| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [7.15](https://github.com/elastic/kibana/pull/109625)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/109625">](https://github.com/elastic/kibana/pull/109625) |
| ✅ |  [7.x](https://github.com/elastic/kibana/pull/109626)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/109626">](https://github.com/elastic/kibana/pull/109626) |

The backport PRs will be merged automatically after passing CI.
